File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        String s1 = s.trim();
        String[] arr = s1.split(" ");
        String s2="";
        
        int left=0;
        int right = arr.length-1;

        for(int i=arr.length-1;i>=0;i--){
         if(arr[i]==""){
                continue;
            }
            s2+=arr[i]+" ";
        }
       
        String result = s2.trim();
        return result;
    }
}

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        reverse(s.begin(),s.end());
        int i=0,idx=0;
        int n=s.size();
        while(i<n){
            while(i<n&&s[i]==' '){
                i++;
            }
            if(i==n) break;
            if(idx!=0) s[idx++]=' ';
            int start=idx;
            while(i<n&&s[i]!=' '){
                s[idx++]=s[i++];
            }
            reverse(s.begin()+start,s.begin()+idx);
        }
        s.resize(idx);
        return s;
    }
};